Assets/HotmaxEditorVR/propertiesMenu/propertiesMenu.cs

```diff
@@ -0,0 +1,207 @@
+                for (int i = 0; i < _selectedObject.GetComponent<Renderer>().materials.Length; i++)
+                {
+                    materialNames.Add(_selectedObject.GetComponent<Renderer>().materials[i].name);
+                }
+            }
+
+            //get all the materials in the children
+            //--TODO: Do this on GLTF serialization
+            Renderer[] renderers = _selectedObject.GetComponentsInChildren<Renderer>();
+            foreach (Renderer renderer in renderers)
+            {
+                materialNames.Add(renderer.material.name);
+            }
+
+            removeLastMaterialPage();
+
+            //get only unique instances of the material names
+            materialNames = materialNames.Distinct().ToList();
+
+            //create "chunks" groups of 4 to display the object
+            chunkedMaterialList = ListExtensions.ChunkBy(materialNames, 4);
+            showMaterialPage(0);
+
+            //--------------------------------------MASS--------------------------------------\\
+
+            //--GETTING
+            //if it has a Rb it should be under 90 mass units
+            if (_selectedObject.GetComponent<Rigidbody>())
+            {
+                selectedObjectRb = _selectedObject.GetComponent<Rigidbody>();
+
+                if (selectedObjectRb.mass < 60)
+                {
+                    physicsText.text = "#grabbable #throwable";
+                }
+                else if (selectedObjectRb.mass > 60 && selectedObjectRb.mass < 90)
+                {
+                    physicsText.text = "#heavy #non-grabbable";
+                }
+
+                slider.transform.localPosition = new Vector3(0, 0, selectedObjectRb.mass / 10);
+            }
+            else
+            {
+                slider.transform.localPosition = new Vector3(0, 0, 10);
+                physicsText.text = "#non-moveable";
+            }
+        }
+    }
+
+    public void materialPageUp()
+    {
+        if (currentMaterialsPage >= chunkedMaterialList.Count - 1)
+        {
+            return;
+        }
+        else
+        {
+            currentMaterialsPage += 1;
+            removeLastMaterialPage();
+            showMaterialPage(currentMaterialsPage);
+        }
+    }
+
+    public void materialPageDown()
+    {
+        if (currentMaterialsPage <= 0)
+        {
+            return;
+        }
+        else
+        {
+            currentMaterialsPage -= 1;
+            removeLastMaterialPage();
+            showMaterialPage(currentMaterialsPage);
+        }
+    }
+
+    public void updatePhysicsInfo()
+    {
+        //TODO: move this to a slider class mutate, these properties and animations will probably have to go through the state
+
+        //--TODO: needs to be a public method for the trigger unclick of the slider
+        if(_selectedObject != null)
+        {
+            sliderUnit = slider.transform.localPosition.z;
```

Comment:
let's make sliderUnit an argument that gets passed from the sliderClass through the UnityEvent... go down to the sliderClass for further instructions.. for now just make sliderUnit a float argument to this method.